crates/client/rpc-core/src/lib.rs

```diff
@@ -36,11 +41,36 @@ pub struct PredeployedAccountWithBalance {
     pub balance: FieldElement,
 }
 
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DeclareV0Result {
+    pub txn_hash: TransactionHash,
+    pub class_hash: ClassHash,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum DeclareV0Transaction {
```

Comment:
```suggestion
pub enum DeclareTransaction {
```

My idea was actually

1. Have a CustomDeclareV0Transaction type like this

```rust
struct CustomDeclareV0Transaction {
  declare_tx: DeclareTransactionV0V1,
  program: Vec<u8>
  ...
```

Since this is our custom type it will mostly work with serde (if it doesn't we can skip). We will use `CustomDeclareV0Transaction` as an input for `declare_v0_contract` RPC.

2. Have a `DeclareV0Transaction ` type inside `rpc` not `rpc-core` as its not needed in RPC. Also we can rename it to `DeclareTransaction` and instead of a tuple for V0 it will contain CustomDeclareV0Transaction